### PR TITLE
T-32182 Migrate zeroex trades

### PR DESCRIPTION
**Pull request by couralex6**
# Thank you for contributing to Spellbook!
Please refer to the top of the `readme` in the root of Spellbook to learn how to contribute to Spellbook on DuneSQL.

### CODE REVIEW

File: models/zeroex/optimism/zeroex_optimism_native_fills.sql [hunk 1]
```diff
@@ -57,21 +57,21 @@ WITH
...
                 END = mp.contract_address
         LEFT OUTER JOIN {{ ref('tokens_erc20') }} mt ON mt.contract_address = fills.makerToken and mt.blockchain = 'optimism'
         LEFT OUTER JOIN {{ ref('tokens_erc20') }} tt ON tt.contract_address = fills.takerToken and tt.blockchain = 'optimism'
          where 1=1 
                 {% if is_incremental() %}
-                AND evt_block_time >= date_trunc('day', now() - interval '1 week')
+                AND evt_block_time >= date_trunc('day', now() - interval '7' day)
                 {% endif %}
                 {% if not is_incremental() %}
                 AND evt_block_time >= '{{zeroex_v3_start_date}}'
```
Comment (Hosuke) [inline models/zeroex/optimism/zeroex_optimism_native_fills.sql:77]: Add missing timestamp.
```suggestion
                AND evt_block_time >= TIMESTAMP '{{zeroex_v3_start_date}}'
```

File: models/zeroex/polygon/zeroex_polygon_native_fills.sql [hunk 2]
```diff
@@ -56,24 +56,24 @@ WITH
             date_trunc('minute', evt_block_time) = tp.minute and tp.blockchain = 'polygon'
             AND CASE
                     -- Set Deversifi ETHWrapper to WETH
-                    WHEN SUBSTRING(fills.takerAssetData,17,20) IN ('0x0000000000000000000000000000000000001010') THEN '0x0d500b1d8e8ef31e21c99d1db9a6444d3adf1270'
+                    WHEN SUBSTRING(fills.takerAssetData,17,20) IN (0x0000000000000000000000000000000000001010) THEN 0x0d500b1d8e8ef31e21c99d1db9a6444d3adf1270
```
Comment (Hosuke) [inline models/zeroex/polygon/zeroex_polygon_native_fills.sql:59]: We probably need to use bytearray_substring() for varbinary typed values.

**Comment by Hosuke** [comment]
The current dbt test failure is caused by the select * in union all (results in order mismatch), I will try to fix tomorrow.

**Review by aalan3 — APPROVED** [review]

**Comment by aalan3** [comment]
this is currently failing due to a issue with trino, we can merge when it's fixed.

**Pull request merged**